astropy/io/fits/convenience.py

```diff
@@ -433,15 +434,24 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
-def table_to_hdu(table):
+def table_to_hdu(table, astropy_native=False):
```

Comment:
The major factor influencing this is backward compatibility. If we make default ``True``, the earlier codes will fail and also other packages dependent on astropy might not want a change in original functionality.